packages/melody-parser/src/TokenStream.js

```diff
@@ -168,7 +168,7 @@ function getAllTokens(lexer, options) {
         ) {
             tokens[tokens.length] = token;
         }
-        acceptWhitespaceControl = true;
+        acceptWhitespaceControl = true && options.ignoreWhitespace;
```

Comment:
```suggestion
        acceptWhitespaceControl = options.ignoreWhitespace;
```

Should be equivalent.